proposals/semi-auto-properties.md

```diff
@@ -10,6 +10,257 @@ Two common scenarios are that you want to apply a constraint on the setter, ensu
+
+```cs
+class C
+{
+    public C() { P = ""; }
+
+    public string P { get => field; set => field = value; }
+}
+```
+
+```cs
+class C
+{
+    public string P { get => field; set => field = value; } = "";
+}
+```
+
+In the same vein as how `var` infers as nullable for reference types, the `field` type should be nullable for reference types. This makes sense of `field ??` as not being followed by dead code, and it avoids producing a misleading warning in the following example:
+
+```cs
+public string AmbientValue
+{
+    get => field ?? parent.AmbientValue;
+    set
+    {
+        if (value == parent.AmbientValue)
+            field = null; // No warning here. Resume following the parent's value.
+        else
+            field = value; // Stop following the parent's value
+    }
+}
+```
+
+Just like with nullability of `var`, it's expected that such patterns will exist with manually-declared backing fields.
+
+To land in this sweet spot implicitly, without having to write an attribute each time, nullability analysis will combine an inherent nullability of the field with the behavior of `[field: NotNull]`. This allows maybe-null assignments without warning, which is desirable as shown above, while simultaneously allowing a scenario like `=> field.Trim();` without requiring an intervention to silence a warning that `field` could be null. Making sure `field` has been assigned is already covered by the warning that ensures non-nullable properties are assigned by the end of each constructor.
+
+### `nameof`
+
+In places where `field` is a keyword (see the [Shadowing](#shadowing) section), `nameof(field)` will fail to compile, like `nameof(nint)`. It is not like `nameof(value)`, which is the thing to use when property setters throw ArgumentException as some do in the .NET core libraries. In contrast, `nameof(field)` has no expected use cases. If it did anything, it would return the string `"field"`, consistent with how `nameof` behaves in other circumstances by returning the C# name or alias, rather than the metadata name.
+
+### Overrides
+
+Overriding properties may use `field`. Such usages of `field` refer to the backing field for the overriding property, separate from the backing field of the base property if it has one. There is no ABI for exposing the backing field of a base property to overriding classes since this would break encapsulation.
+
+Like with regular auto properties, semi-auto properties that override a base property must override all accessors ([LDM decision](https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-05-02.md#partial-overrides-of-virtual-properties)).
+
+### Shadowing
+
+`field` can be shadowed by parameters or locals in a nested scope ([LDM decision](https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-02-16.md#open-questions-in-field)). Since `field` represents a field in the type, even if anonymously, the shadowing rules of regular fields should apply.
```

Comment:
oh.  this feels like soemthign that deviates from how i remember, or how i think we may have talked about thsi recently.  i *thought* we were going with an approach that since it is a keyword now in a property setting, that you just literally could not write a parameter/local called `field`.  So no shadowing cases exist.

Note: this depends on us going with the 'breaking change' approach here.